proposals/nnnn-button_subscription_response_from_hmi.md

```diff
@@ -0,0 +1,67 @@
+# Button Subscription response from HMI
+
+* Proposal: [SDL-NNNN](nnnn-button_subscription_response_from_hmi.md)
+* Author: [LuxoftAKutsan](https://github.com/LuxoftAKutsan)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core]
+
+## Introduction
+This proposal is to replace `OnButtonSubscription` notification
+with `SubscribeButton` request/response,`UnsubscribeButton` request/response to HMI. 
+
+## Motivation
+
+
+In the current design when an app wants to subscribe to particular button, the app sends a SubscribeButton request to SDL Core. SDL Core then immediately provides a response of 'SUCCESS' to the app and sends the OnButtonSubscription notification to HMI. The problem with this design is that if for any reason HMI does not accept or fails to accept this subscription then HMI has no way to notify the app that the button has not been successfully subscribed. This results in the app assuming that it has successfully subscribed to a button when it has not.
+Through this proposal we propose a design for fixing this problem.
+The proposed design is similar to how other subscriptions such as VehicleData, InteriorVehicleData work on the system.
+
+
+## Proposed solution
+
+### API changes
+
+Remove `<function name="OnButtonSubscription" messagetype="notification">`. 
+
+Add new RPCs to HM_API:
+```xml
+  <function name="SubscribeButton" messagetype="request">
+        <description>
+            Subscribes/Unsubscribes to/from buttons.            
+        </description>
+		
+	    <param name="appID" type="Integer" mandatory="true">
+			<description>The ID of the application requesting this button-subscription status change.</description>
+        </param>
+		
+        <param name="buttonName" type="ButtonName" mandatory="true">
+            <description>Name of the button to subscribe/unsubscribe.</description>
+        </param>
+		
+		<param name="subscribe" type="Boolean" mandatory="true">
+			<description>Flag to subscribe/unsubscribe to button presses.If true, the requester will be subscribed. If false, the requester will not be subscribed and be removed as a subscriber if it was previously subscribed.</description>
+		</param>
+    </function>
+    
+    <function name="SubscribeButton" messagetype="response">
+    </function>
+```
+
+ - After sending a SubscribeButton request, if the HMI does not respond within DefaultTimeout period, then SDL Core shall provide a response of TIMED_OUT to the requesting application.
+ - If after SDL Core send a 'TIMED_OUT' response to the application, HMI sends the response to core notifying that the button has been successfully subscribed, then SDL Core shall send a request to HMI to unsubscribe the button. This is done to ensure the application and HMI are in sync.
```

Comment:
A few updates:
- "SDL Core send" should be "SDL Core sends"
- 'TIMED_OUT' should be `TIMED_OUT` for proper codeblock formatting
- "core" should be capitalized (...response to Core notifying...)